crates/ty_project/src/db.rs

```diff
@@ -92,6 +92,15 @@ impl ProjectDatabase {
         self.project().check(self, reporter)
     }
 
+    /// Checks all files in the workspace, ignoring open files setting.
+    ///
+    /// This is used to power the workspace diagnostics capability in the language server.
+    pub fn check_all_files(&self) -> Vec<Diagnostic> {
```

Comment:
I think I'd prefer not to introduce this new method and instead:

a) Remove the behavior that open files has today and make `check` always check all files (the LSP can call `check_file` if it only wants diagnostics for open files). The downside of this is that we may need another way to surface setting diagnostics
b) Add a `CheckMode` field

Reply:
I actually thought of changing the current logic something like (b). I don't think we should do (a) as the playground checks only the files that are marked open.

Reply:
Oh good point. I think that'll require https://github.com/astral-sh/ty/issues/619 because currently the server uses the `check_file` method directly but to include the virtual files in the workspace diagnostics, we need to know the set of virtual files that are available. After #619, we can get the `open_files` set and filter based on virtual path.

Reply:
Or, we could just ask the server index to get all the open virtual files.

Reply:
> Or, we could just ask the server index to get all the open virtual files.

I think that would work for workspace diagnostics but not necessarily how we use `open_files` in `check_file_impl` to determine if the file is open.

Reply:
Something else we would have to be careful about when moving it to `System` is that we can't use it in any salsa query because the result can change from the outside.